src/pages/sales/bulkOrders/BulkOrderDetails.tsx

```diff
@@ -19,16 +28,48 @@ const BulkOrderDetails = () => {
   const [salesOrders, setSalesOrders] = useState<SalesOrder[]>([]);
   const [alert, setAlert] = useState<AlertType | null>(null);
   const [loading, setLoading] = useState<boolean>(true);
+  const [activeStep, setActiveStep] = useState<number>(0);
+
+  const statusStepper = useMemo(
```

Comment:
why duplicated code? use a callback to set the current bulkOrderStep object into a state, then use that state in line 126 instead of repeating the code